gprofiler/application_separators.py

```diff
@@ -0,0 +1,193 @@
+        return None
+
+
+class UwsgiApplicationSeparator(ApplicationSeparator):
+    @property
+    def app_prefix(self) -> str:
+        return "uwsgi"
+
+    def is_supported(self, process: Process) -> bool:
+        return process.cmdline()[0] == "uwsgi"
+
+    @staticmethod
+    def _find_wsgi_from_config_file(process: Process) -> Optional[str]:
+        config_file = None
+        for arg in process.cmdline():
+            if arg.endswith(".ini"):
+                config_file = arg
+
+        if config_file is None:
+            return None
+
+        if not os.path.isabs(config_file):
+            config_file = os.path.join(process.cwd(), config_file)
+
+        config = configparser.ConfigParser()
+        config.read(resolve_host_path(process, config_file))
+        try:
+            return config.get("uwsgi", "module")
+        except (configparser.NoSectionError, configparser.NoOptionError):
+            pass
+
+        return None
+
+    def get_application_name(self, process: Process) -> Optional[str]:
+        wsgi = get_positional_arg(process.cmdline(), "-w")
+        if wsgi is not None:
+            return wsgi
+
+        wsgi = self._find_wsgi_from_config_file(process)
+        if wsgi is not None:
+            return wsgi
+
+        logger.warning("Couldn't find uwsgi wsgi module, both from cmdline and from config file")
+        return None
+
+
+class CeleryApplicationSeparator(ApplicationSeparator):
+    @property
+    def app_prefix(self) -> str:
+        return "celery"
+
+    def is_supported(self, process: Process) -> bool:
+        if process.cmdline()[0] == "celery":
+            return True
+
+        return len(process.cmdline()) >= 3 and ["-m", "celery"] == process.cmdline()[1:]
+
+    def get_application_name(self, process: Process) -> Optional[str]:
+        app_name = get_positional_arg(process.cmdline(), "-A") or get_positional_arg(process.cmdline(), "--app")
+        if app_name is None:
+            logger.warning("Couldn't find positional argument -A or --app for application indication")
+            return None
+
+        return app_name
+
+
+class PySparkApplicationSeparator(ApplicationSeparator):
+    @property
+    def app_prefix(self) -> str:
+        return "pyspark"
+
+    def is_supported(self, process: Process) -> bool:
+        # We're looking for pythonXX -m pyspark.daemon
+        return process.cmdline() >= 3 and ["-m", "pyspark.daemon"] == process.cmdline()[1:]
+
+    def get_application_name(self, process: Process) -> Optional[str]:
+        return "generic pyspark.daemon"
+
+
+class PythonModuleApplicationSeparator(ApplicationSeparator):
+    @property
+    def app_prefix(self) -> str:
+        return "generic-python"
+
+    def is_supported(self, process: Process) -> bool:
+        if "python" not in process.cmdline()[0]:
+            return False
+
+        return (len(process.cmdline()) > 3 and process.cmdline()[1] == "-m") or (
```

Comment:
Shouldn't it be
```suggestion
        return (len(process.cmdline()) >= 3 and process.cmdline()[1] == "-m") or (
```
?